src/bp/ui-shared/src/MainContent/Header/typings.d.ts

```diff
@@ -0,0 +1,13 @@
+import { IButtonProps, ITabProps, IconName } from '@blueprintjs/core'
+
+export interface HeaderProps {
+  tabs?: ITabProps[]
+  tabChange?: function
+  buttons?: HeaderButtonProps[]
+}
+
+export interface HeaderButtonProps {
+  onClick: function
```

Comment:
I think we should avoid generic methods as much as possible, so either use like `onClick: (e: React...Event) => void` or `onClick: () =>void` if the element is not important